ngraph/core/src/op/util/recurrent_sequence.cpp

```diff
@@ -0,0 +1,62 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "recurrent_sequence.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+void op::util::validate_seq_input_rank_dimension(const std::vector<ngraph::PartialShape>& input)
+{
+    // Verify static ranks for all inputs
+    for (size_t i = 0; i < input.size(); i++)
+    {
+        NGRAPH_CHECK((input[i].rank().is_static()),
+                     "RNN Sequence supports only static rank for input tensors.");
+    }
+
+    for (size_t i = 0; i < input.size(); i++)
+    {
+        if (i == input.size() - 1)
```

Comment:
It would be much more readable and verbose if you would have ie:
```
enum {
  X,
  INITIAL_H,
  SEQ_LEN,
  ...
};
```
and use those values instead of subtracting from vector size some magic number.

Reply:
Good idea, already added.